searx/plugins/tor_check.py

```diff
@@ -0,0 +1,61 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""
+A plugin to check if the ip address of the request is a TOR exit node
+if the user searches for "tor" "tor check" or "check tor"
+It fetches the tor exit node list from https://check.torproject.org/exit-addresses
+and parses all the IPs into a list, then checks if the user's ip address
+is in it.
+
+Enable in 'settings.yml' (uncomment "Tor check plugin"
+in the 'enabled_plugins' section)
+"""
+
+import re
+from urllib.request import urlopen
+from urllib.error import HTTPError
+
+name = "Tor check plugin"
+description = """
+This plugin checks if the address of the request is a TOR exit node,
+and informs the user if it is, like check.torproject.org but from searxng.
+"""
+default_on = False
+
+# Regex for exit node addresses in the list.
+reg = re.compile(r"(?<=ExitAddress )\S+")
+
+
+def post_search(request, search):
```

Comment:
See `self_info.py` :
https://github.com/searxng/searxng/blob/44f2eb50a5566e8e14a1e970f1ed775bf025c1ea/searx/plugins/self_info.py#L35-L36

so the plugin is triggered only on the first page.